app/models/sitewide_setting.rb

```diff
@@ -12,4 +11,20 @@ def name
     key.humanize
   end
 
+  def self.set(key, on)
+    flag = find_by_key_or_create(key)
+    flag.update(on: on)
+  end
+
+  def self.on?(name)
+    if flag = find_by(key: name)
+      flag.on
+    else
+      false
+    end
+  end
+
+  def self.find_by_key_or_create(key)
```

Comment:
As noted above, you don't need to define this method, ActiveRecord already gives you `find_or_create_by`.